roofit/roofitcore/src/RooLinearVar.cxx

```diff
@@ -248,9 +248,7 @@ std::list<std::string> RooLinearVar::getBinningNames() const
 {
   std::list<std::string> binningNames(1, "");
 
-  RooFIter iter = _altBinning.fwdIterator();
-  const RooAbsArg* binning = 0;
-  while((binning = iter.next())) {
+  for (auto const* binning : static_range_cast<RooAbsArg*>(_altBinning)) {
```

Comment:
```suggestion
  for (RooAbsArg const* binning : _altBinning) {
```
No need to cast here, the class is already RooAbsArg (maybe replace the `auto` with RooAbsArg too such that it's explicitly clear from now).